### ISSUE TITLE
ipython 8.2.0 mixing history from multiple ipython sessions

### ISSUE THREAD

**Issue by segasai** [ipython/ipython#13631]
Hi, 

After updating to ipython 8.2.0 I observe a very annoying behaviour, when pressing 'UP_ARROW' in long running ipython session brings not the previous command from history, but a command from some other ipython session running concurrently. 

How to reproduce.
* Start ipython session 1
* Start ipython session 2
* Type x=1 in session 1
* Type y=1 in session 2
* Type x=2 in session 1
* Type y=2 in session 2
* Now in session 1 press 'UP_ARROW', and it will bring y=1 instead of x=2 

This is very inconvenient behaviour that completely breaks the workflow in long running ipython sessions. I assume it must be a regression.

Thank you.

(I'm using python 3.8 with the latest prompt-toolkit 3.0.29 on linux (ubuntu 18.04)

**Comment by segasai** [comment]
I have just checked that this was caused by the https://github.com/ipython/ipython/pull/13592 as the problem didn't occur before it.

**Comment by ljames8** [comment]
Hi
I have the exact same issue on macOS 12.1 with python 3.9.10 and prompt-toolkit 3.0.8

**Comment by anwarnunez** [comment]
This issue is indeed very inconvenient. I'm not sure the current behaviour is ever desirable.

**Comment by Carreau** [comment]
closed by #13657

**Comment by lukelbd** [comment]
I'm still experiencing this issue in IPython 8.3.0 and IPython 8.4.0 (downloaded with conda on linux). Tested with `conda install ipython==8.3.0` and `conda install ipython==8.4.0`.

**Comment by hristog** [comment]
I'm still experiencing this issue on: IPython 8.5.0 (and the versions all the way down to 8.2.0, inclusive), and Python 3.8.5.
I've had to downgrade to IPython 8.1.0, in order to have the up-arrow-key history functionality work properly.

**Comment by harenbrs** [comment]
@hristog Can confirm, seeing the same on IPython 8.5.0:

```
Python 3.8.6 (default, Nov  9 2020, 16:24:25) 
Type 'copyright', 'credits' or 'license' for more information
IPython 8.5.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: 42
Out[1]: 42
                                                                                                                                                                            0.013377s
In [2]: %history
42
%history
                                                                                                                                                                            0.000982s
In [3]: X('year').get_previous_start_time(dt)  # up-arrow
```

**Comment by hristog** [comment]
Thanks for confirming, @harenbrs!

**Comment by derVedro** [comment]
same with  IPython 8.6.0, fixed it by updating old **promt_toolkit** (3.0.5) to the latest one (3.0.32)
